fastlane/lib/fastlane/actions/slack.rb

```diff
@@ -185,99 +292,13 @@ def self.details
-            title: k.to_s,
-            value: Slack::Notifier::Util::LinkFormatter.format(v.to_s),
-            short: false
-          }
-        end
-
-        # Add the lane to the Slack message
-        # This might be nil, if slack is called as "one-off" action
-        if should_add_payload[:lane] && Actions.lane_context[Actions::SharedValues::LANE_NAME]
-          slack_attachment[:fields] << {
-            title: 'Lane',
-            value: Actions.lane_context[Actions::SharedValues::LANE_NAME],
-            short: true
-          }
-        end
-
-        # test_result
-        if should_add_payload[:test_result]
-          slack_attachment[:fields] << {
-            title: 'Result',
-            value: (options[:success] ? 'Success' : 'Error'),
-            short: true
-          }
-        end
-
-        # git branch
-        if Actions.git_branch && should_add_payload[:git_branch]
-          slack_attachment[:fields] << {
-            title: 'Git Branch',
-            value: Actions.git_branch,
-            short: true
-          }
-        end
-
-        # git_author
-        if Actions.git_author_email && should_add_payload[:git_author]
-          if FastlaneCore::Env.truthy?('FASTLANE_SLACK_HIDE_AUTHOR_ON_SUCCESS') && options[:success]
-            # We only show the git author if the build failed
-          else
-            slack_attachment[:fields] << {
-              title: 'Git Author',
-              value: Actions.git_author_email,
-              short: true
-            }
-          end
-        end
-
-        # last_git_commit
-        if Actions.last_git_commit_message && should_add_payload[:last_git_commit]
-          slack_attachment[:fields] << {
-            title: 'Git Commit',
-            value: Actions.last_git_commit_message,
-            short: false
-          }
-        end
-
-        # last_git_commit_hash
-        if Actions.last_git_commit_hash(true) && should_add_payload[:last_git_commit_hash]
-          slack_attachment[:fields] << {
-            title: 'Git Commit Hash',
-            value: Actions.last_git_commit_hash(short: true),
-            short: false
-          }
-        end
-
-        # merge additional properties
-        deep_merge(slack_attachment, options[:attachment_properties])
+      def self.trim_message(message)
+        Runner.trim_message(message)
       end
 
-      # Adapted from https://stackoverflow.com/a/30225093/158525
-      def self.deep_merge(a, b)
-        merger = proc do |key, v1, v2|
-          Hash === v1 && Hash === v2 ?
-                 v1.merge(v2, &merger) : Array === v1 && Array === v2 ?
-                   v1 | v2 : [:undefined, nil, :nil].include?(v2) ? v1 : v2
-        end
-        a.merge(b, &merger)
+      def self.generate_slack_attachments(options)
+        UI.deprecation('`Fastlane::Actions::Slack.generate_slack_attachments` is subject to be removed as Slack recommends migrating `attachments` to Block Kit. fastlane will also the same direction.')
```

Comment:
```suggestion
        UI.deprecated('`Fastlane::Actions::Slack.generate_slack_attachments` is subject to be removed as Slack recommends migrating `attachments` to Block Kit. fastlane will also the same direction.')
```
`UI.deprecation` does not exist and **halt fastlane**
<img width="1233" alt="Screenshot 2021-04-11 at 09 41 41" src="https://user-images.githubusercontent.com/5364500/114296340-1a85e600-9aab-11eb-8070-7e6e66433b67.png">

------

Another thought, 
**For now,** can we just use here `UI.important` for this warning because Slack is not deprecating attachments anytime soon. 
Once we will have Block-Kit implementation then we can make it `UI.deprecated` later.


<img width="738" alt="Screenshot 2021-04-11 at 08 30 31" src="https://user-images.githubusercontent.com/5364500/114296374-69cc1680-9aab-11eb-9bb4-2a7f1fba535e.png">

Reply:
> UI.deprecation does not exist and halt fastlane

Nice catch! Thanks🙈 

> For now, can we just use here UI.important for this warning because Slack is not deprecating attachments anytime soon.

Yes, It's not deprecated yet on _Slack_. I meant the use of this method is deprecated in fastlane or plugins as I feel like fastlane has nothing to guarantee this interface or use of `attachments` in JSON for the future. If we postpone deprecating this until we get Block Kit implementation, we need to end up leaving the old implementation even though there's no internal code to use it. That's unacceptable, I think.